jetty-server/src/main/java/org/eclipse/jetty/server/handler/InetAccessHandler.java

```diff
@@ -26,128 +26,136 @@
     {
-        _addrs.clear();
-        _names.clear();
+        _set.clear();
     }
 
     /**
-     * Includes an InetAddress pattern
+     * Includes an InetAccess pattern with an optional connector name, address and URI mapping.
+     *
+     * <p>The connector name is separated from the InetAddress pattern with an '@' character,
+     * and the InetAddress pattern is separated from the URI pattern using the "|" (pipe)
+     * character. URI patterns follow the servlet specification for simple * prefix and
+     * suffix wild cards (e.g. /, /foo, /foo/bar, /foo/bar/*, *.baz).</p>
+     *
+     * <br>Examples:
+     * <ul>
+     * <li>"connector1@127.0.0.1|/foo"</li>
+     * <li>"127.0.0.1|/foo"</li>
+     * <li>"connector1@127.0.0.1"</li>
+     * <li>"127.0.0.1"</li>
+     * </ul>
      *
-     * @param pattern InetAddress pattern to include
+     * @param pattern InetAccess pattern to include
      * @see InetAddressSet
      */
     public void include(String pattern)
     {
-        _addrs.include(pattern);
+        _set.include(PatternTuple.from(pattern));
     }
 
     /**
-     * Includes InetAddress patterns
+     * Includes InetAccess patterns
      *
      * @param patterns InetAddress patterns to include
      * @see InetAddressSet
      */
     public void include(String... patterns)
     {
-        _addrs.include(patterns);
+        for (String pattern : patterns)
+            include(pattern);
     }
 
     /**
-     * Excludes an InetAddress pattern
-     *
-     * @param pattern InetAddress pattern to exclude
-     * @see InetAddressSet
+     * Includes an InetAccess entry.
+     * @param connectorName optional name of a connector to include.
+     * @param addressPattern optional InetAddress pattern to include.
+     * @param pathSpec optional pathSpec to include.
      */
-    public void exclude(String pattern)
+    public void include(String connectorName, String addressPattern, PathSpec pathSpec)
     {
-        _addrs.exclude(pattern);
+        _set.include(new PatternTuple(connectorName, InetAddressPattern.from(addressPattern), pathSpec));
     }
 
     /**
-     * Excludes InetAddress patterns
+     * Excludes an InetAccess entry pattern with an optional connector name, address and URI mapping.
      *
-     * @param patterns InetAddress patterns to exclude
-     * @see InetAddressSet
-     */
-    public void exclude(String... patterns)
-    {
-        _addrs.exclude(patterns);
-    }
-
-    /**
-     * Includes a connector name.
+     * <p>The connector name is separated from the InetAddress pattern with an '@' character,
+     * and the InetAddress pattern is separated from the URI pattern using the "|" (pipe)
+     * character. URI patterns follow the servlet specification for simple * prefix and
+     * suffix wild cards (e.g. /, /foo, /foo/bar, /foo/bar/*, *.baz).</p>
      *
-     * @param name Connector name to include in this handler.
-     */
-    public void includeConnector(String name)
```

Comment:
I think the `includeConnector` and `excludeConnector` methods should be retained as deprecated methods.

`excludeConnector(name)` can easily be implemented as `exclude(name+"@")`

Unfortunately `includeConnector(name)` can not so easily be implemented.  If there are no current inclusions, then  `include(name+"@")` works, but if includes exist or a are subsequently added, then the meaning will change.  So I think  throwing unsupported operation is probably best